frontend/src/ui/NewWindow/LegacyNewWindow.js

```diff
@@ -2,195 +2,234 @@ import React from 'react'
-     if (this.window) {
-       if (this.props.closeOnUnmount) {
-         this.window.close()
-       } else if (this.props.children) {
-         // Clone any children so they aren't removed when react stops rendering
-         const clone = this.container.cloneNode(true)
-         clone.setAttribute('id', 'new-window-container-static')
-         this.window.document.body.appendChild(clone)
-       }
-     }
-   }
-
-   /**
-    * Release the new window and anything that was bound to it.
-    */
-   release () {
-     // This method can be called once.
-     if (this.released) {
-       return
-     }
-     this.released = true
-
-     // Remove checker interval.
-     clearInterval(this.windowCheckerInterval)
-
-     // Call any function bound to the `onUnload` prop.
-     const { onUnload } = this.props
-
-     if (typeof onUnload === 'function') {
-       onUnload(null)
-     }
-   }
+  /**
+   * The NewWindow function constructor.
+   * @param {Object} props
+   */
+  constructor(props) {
+    super(props)
+    this.container = null
+    this.window = null
+    this.windowCheckerInterval = undefined
+    this.released = false
+    this.state = {
+      localShowPrompt: false,
+      mounted: false
+    }
+    this.beforeUnloadListener = this.beforeUnloadListener.bind(this)
+  }
+
+  /**
+   * @override
+   */
+  componentDidMount() {
+    const { onChangeFocus } = this.props
+
+    this.openChild()
+    this.setState({ mounted: true })
+
+    this.window?.addEventListener('beforeunload', this.beforeUnloadListener, { capture: true })
+
+    this.window?.addEventListener('blur', () => {
+      onChangeFocus('hidden')
+    })
+    this.window?.addEventListener('focus', () => {
+      onChangeFocus('visible')
+    })
+  }
+
+  /**
+   *
+   * @override
+   */
+  componentDidUpdate(prevProps) {
+    const { doFocus, showPrompt } = this.props
+    const { localShowPrompt } = this.state
+    if (prevProps.doFocus !== doFocus && doFocus) {
+      this.window?.focus()
+    }
+
+    if (localShowPrompt !== showPrompt) {
+      this.setState({ localShowPrompt: showPrompt })
+    }
+  }
+
+  /**
+   * Closes the opened window (if any) when NewWindow will unmount if the
+   * prop {closeOnUnmount} is true, otherwise the NewWindow will remain open
+   * @override
+   */
+  componentWillUnmount() {
+    const { children, closeOnUnmount } = this.props
+    if (this.window) {
+      if (closeOnUnmount) {
+        this.window.close()
+      } else if (children) {
+        // Clone any children so they aren't removed when react stops rendering
+        const clone = this.container?.cloneNode(true)
+        clone?.setAttribute('id', 'new-window-container-static')
+        this.window.document.body.appendChild(clone)
+      }
+    }
+    this.window?.removeEventListener('beforeunload', this.beforeUnloadListener, { capture: true })
+  }
+
+  beforeUnloadListener(e) {
+    e.preventDefault()
+    const { localShowPrompt } = this.state
```

Comment:
pourquoi ne pas utiliser directement `this.props.showPrompt`?

Reply:
effectivement, un reliquat d'un truc que je testais avant. je clean ça